src/main/java/com/gooddata/warehouse/WarehouseService.java

```diff
@@ -492,56 +494,104 @@ public WarehouseS3Credentials getWarehouseS3Credentials(final Warehouse warehous
     /**
      * add new S3 credentials to the Warehouse
      *
-     * @param warehouse     warehouse to add credentials to
+     * @param warehouse     warehouse the S3 credentials should be added to
      * @param s3Credentials the credentials to store
      * @return added credentials (not null)
      */
-    public FutureResult<WarehouseS3Credentials> addS3CredentialsToWarehouse(final Warehouse warehouse,
-                                                                            final WarehouseS3Credentials s3Credentials) {
+    public FutureResult<WarehouseS3Credentials> addS3Credentials(final Warehouse warehouse,
+                                                                 final WarehouseS3Credentials s3Credentials) {
         notNull(warehouse, "warehouse");
         notEmpty(warehouse.getId(), "warehouse.id");
         notNull(s3Credentials, "s3Credentials");
 
-        final WarehouseTask task;
+        final WarehouseS3Credentials s3CredentialsWithLinks = s3Credentials.withLinks(warehouse.getId());
```

Comment:
the aforementioned change has been reverted (misunderstood the suggestion). Passing `targetUri` argument now, `.getListUri()` for `POST`, `.getUri()` for `PUT` and `DELETE`.